vulcanai/models/basenetwork.py

```diff
@@ -665,14 +665,16 @@ def fit(self, train_loader, val_loader, epochs,
                     self.lr_scheduler.step(epoch=epoch)
 
                 if self.early_stopping:
-                    if self.early_stopping_metric == "loss":
-                        if valid_loss:
+
+                    # only call if you've actually measured these.
+                    if valid_loss:
+                        if self.early_stopping_metric == "loss":
                             valid_loss_neg = -1 * valid_loss
```

Comment:
you're negating this because early stoping will look for the largest number consistently?

Reply:
yes

Reply:
Can we add a comment about that? We’ll continue to add more functionality to be based off of other metrics in the future and this will have to be kept in mind

Reply:
ok done.